src/components/MermaidDash/MermaidDash.jsx

```diff
@@ -64,30 +73,36 @@ const MermaidDash = () => {
           }
 
           const data = await response.json()
-          setProjectData((prevData) => {
```

Comment:
we were previously setting react state that renders many many things within a while loop. Now we are just setting some state that impacts the loader and nothing else.